lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java

```diff
@@ -1221,6 +1228,110 @@ private Location createListingLocation(String path)
         return createLocation(path);
     }
 
+    @Test
+    void testFileDoesNotExistUntilClosed()
+            throws Exception
+    {
+        Location location = getRootLocation().appendPath("testFile2");
+        getFileSystem().deleteFile(location);
+        try (OutputStream out = getFileSystem().newOutputFile(location).create()) {
+            assertFalse(fileExistsInListing(location));
+            assertFalse(fileExists(location));
+            out.write("test".getBytes(UTF_8));
+            assertFalse(fileExistsInListing(location));
+            assertFalse(fileExists(location));
+        }
+        assertTrue(fileExistsInListing(location));
+        assertTrue(fileExists(location));
+        getFileSystem().deleteFile(location);
+    }
+
+    @Test
+    public void testLargeFileDoesNotExistUntilClosed()
+            throws IOException
+    {
+        Location location = getRootLocation().appendPath("testFile3");
+        getFileSystem().deleteFile(location);
+        byte[] bytes = new byte[8192];
+        for (int i = 0; i < bytes.length / 4; i++) {
+            //noinspection NumericCastThatLosesPrecision
+            bytes[i] = (byte) i;
+        }
```

Comment:
I don't follow what is happening here.
Why are we filling only a quarter of the array?

This snippet is used in another method below as well - extract in a helper method.

Reply:
Maybe the hint from https://github.com/trinodb/trino/pull/20800#discussion_r1500591299 was not correctly applied?

Reply:
Seems like the hint was correct: the contents don't matter as long as they differ from the original contents - +1 on extracting, will do. lmk if you have any concern w any of this.